core/src/main/java/com/microsoft/applicationinsights/channel/concrete/ATelemetryChannel.java

```diff
@@ -0,0 +1,390 @@
+                throttling,
+                maxInstantRetries);
+    }
+
+    /**
+     * This Ctor will query the 'namesAndValues' map for data to initialize itself
+     * It will ignore data that is not of its interest, this Ctor is useful for
+     * building an instance from configuration
+     *
+     * @param namesAndValues - The data passed as name and value pairs
+     */
+    public ATelemetryChannel(Map<String, String> namesAndValues) {
+        boolean developerMode = false;
+        String endpointAddress = null;
+        int maxInstantRetries = DEFAULT_MAX_INSTANT_RETRY;
+
+        LimitsEnforcer maxTelemetryBufferCapacityEnforcer = createDefaultMaxTelemetryBufferCapacityEnforcer(null);
+
+        LimitsEnforcer sendIntervalInSecondsEnforcer = createDefaultSendIntervalInSecondsEnforcer(null);
+
+        boolean throttling = true;
+        if (namesAndValues != null) {
+            throttling = Boolean.valueOf(namesAndValues.get(THROTTLING_ENABLED_NAME));
+            developerMode = Boolean.valueOf(namesAndValues.get(DEVELOPER_MODE_NAME));
+            try {
+                String instantRetryValue = namesAndValues.get(INSTANT_RETRY_NAME);
+                if (instantRetryValue != null) {
+                    maxInstantRetries = Integer.parseInt(instantRetryValue);
+                }
+
+            } catch (NumberFormatException e) {
+                InternalLogger.INSTANCE.error("Unable to parse configuration setting %s to integer value.%nStack Trace:%n%s", INSTANT_RETRY_NAME, ExceptionUtils.getStackTrace(e));
+            }
+
+            if (!developerMode) {
+                developerMode = Boolean.valueOf(System.getProperty(DEVELOPER_MODE_SYSTEM_PROPRETY_NAME));
+            }
+            endpointAddress = namesAndValues.get(ENDPOINT_ADDRESS_NAME);
+
+            maxTelemetryBufferCapacityEnforcer
+                    .normalizeStringValue(namesAndValues.get(MAX_TELEMETRY_BUFFER_CAPACITY_NAME));
+            sendIntervalInSecondsEnforcer
+                    .normalizeStringValue(namesAndValues.get(FLUSH_BUFFER_TIMEOUT_IN_SECONDS_NAME));
+        }
+
+        String maxTransmissionStorageCapacity =
+                namesAndValues.get(MAX_TRANSMISSION_STORAGE_CAPACITY_NAME);
+        initialize(
+                endpointAddress,
+                maxTransmissionStorageCapacity,
+                developerMode,
+                maxTelemetryBufferCapacityEnforcer,
+                sendIntervalInSecondsEnforcer,
+                throttling,
+                maxInstantRetries);
+    }
+
+    protected synchronized void initialize(String endpointAddress, String maxTransmissionStorageCapacity,
+                                       boolean developerMode, LimitsEnforcer maxTelemetryBufferCapacityEnforcer,
+                                       LimitsEnforcer sendIntervalInSeconds, boolean throttling, int maxInstantRetry) {
+        makeSureEndpointAddressIsValid(endpointAddress);

```

Comment:
adding `@throws` (it does say in the javadoc). if it throws, it will escape the `TelemetryChannelBase` constructor and end up here:
https://github.com/Microsoft/ApplicationInsights-Java/blob/61ed8ac512838720e2066c3870661550c6ee6a09/core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java#L159

InProcessTelemetryChannel did this, so it should work with the starter as well.